From 612daf0ca5ec5171ecfc0f845807ec1d3bd0f762 Mon Sep 17 00:00:00 2001
From: Jiani Liu <quic_jianil@quicinc.com>
Date: Thu, 6 Apr 2023 17:20:33 +0800
Subject: [PATCH] ath9k: fix incorrect regulatory

ath9k's local regulatory is not updated correctly when user set a
new regulatory, which may cause getting wrong power from CTL table.

Change-Id: I64a4df2d9e06e2d2ac52221264170faa456b0cdf
Signed-off-by: Miaoqing Pan <miaoqing@codeaurora.org>
Signed-off-by: Jiani Liu <quic_jianil@quicinc.com>

diff --git a/drivers/net/wireless/ath/ath9k/init.c b/drivers/net/wireless/ath/ath9k/init.c
index b8747da..5388a0c 100644
--- a/drivers/net/wireless/ath/ath9k/init.c
+++ b/drivers/net/wireless/ath/ath9k/init.c
@@ -1081,6 +1081,10 @@ int ath9k_init_device(u16 devid, struct ath_softc *sc,
 	if (error)
 		goto deinit;
 
+#ifdef CPTCFG_ATH_USER_REGD
+	sc->hw->wiphy->reg_notifier = ath9k_reg_notifier;
+#endif
+
 	reg = &common->regulatory;
 
 	/* Setup TX DMA */
-- 
2.7.4

